kernel/interrupts/src/aarch64/mod.rs

```diff
@@ -217,6 +217,23 @@ pub fn init_pl011_rx_interrupt() -> Result<(), &'static str> {
     int_ctrl.set_destination(PL011_RX_SPI, Some(current_cpu()), u8::MAX)
 }
 
+pub fn init_pci_interrupts(handler: InterruptHandler) -> Result<(), &'static str> {
```

Comment:
Need some docs about what this function does. I presume it registers the given `handler` to all legacy PCI interrupt vectors, but it's better to be clear.

Also, on that note, do we always want the same single interrupt handler to be registered for all PCI interrupts? It's probably better to allow the caller to specify individual handlers for each of the 4 legacy interrupts, e.g., 

```rust
pub fn init_pci_interrupts(
    pin_a_handler: InterruptHandler,
    pin_b_handler: InterruptHandler,
    pin_c_handler: InterruptHandler,
    pin_d_handler: InterruptHandler,
) -> Result<(), &'static str> {
```